### ISSUE TITLE
New Reek violations may not be reported

### ISSUE THREAD

**Issue by mjacobus** [prontolabs/pronto-reek#20]
Given I have the code on GH (`path/some_class.rb`)

```ruby
class SomeClass
  def do_stuff
    One.new
    Two.new
    Three.new
    Four.new
    Five.new
  end
end
```

The current reek status is:

```bash
Inspecting 1 file(s):
.
```

I change the file and create a pull request that changes the file like follows

```ruby
class SomeClass
  def do_stuff
    One.new
    Two.new
    Three.new
    Four.new
    Five.new
    Six.new
  end
end
```

```bash
Inspecting 1 file(s):
S

/path/pronto_reek.rb -- 1 warning:
  [2]:TooManyStatements: SomeClass#do_stuff has approx 6 statements [https://github.com/troessner/reek/blob/master/docs/Too-Many-Statements.md]
```

Pronto rubocop will not create any comment on my PR, because the report is on
the method on line 2, which is not part of the changed code.

Now, if I add a comment in the line of the method name (where reek reports the
violaiton) and update the PR, I will get a comment from pronto, because line num 2 changed.

```ruby
class SomeClass
  def do_stuff # now it reports
    One.new
    Two.new
    Three.new
    Four.new
    Five.new
    Six.new
  end
end
```

It is a bug, isn't it? The new code created a violation in the existing code.

**Comment by grantspeelman** [comment]
+1 to this idea/bug. Would be great if pronto-reek could cater for this somehow.

**Comment by mjacobus** [comment]
I suppose a potential valid approach for a fix would be to compare the violations of the files before the PR changes with the ones gotten from each commit. The new violations should be the ones introduced by the PR.

**Comment by grantspeelman** [comment]
yes, but what line would it put the comment on? what happens if the correct line becomes available later on the PR later, Will it move the comment or leave it as it?

**Comment by mjacobus** [comment]
Good questions. Perhaps it would not comment on the diff line, if not possible but in the PR regular thread. Perhaps there is no perfect solution.

**Comment by ashkulz** [comment]
This is similar to prontolabs/pronto-rubocop#46 -- I suppose a common solution will work for both 🤔
